silkworm/core/crypto/rmd160.c

```diff
@@ -362,13 +362,13 @@ static inline uint32_t load32(const void* src) {
     return w;
 }
 
-void silkworm_rmd160(uint8_t out[20], const uint8_t* ptr, size_t len) {
+void silkworm_rmd160(uint8_t out[20], const uint8_t* ptr, uint32_t len) {
     uint32_t buf[160 / 32];
 
     rmd160_init(buf);
 
     uint32_t current[16];
-    for (size_t remaining = len; remaining >= 64; remaining -= 64) {
+    for (unsigned remaining = len; remaining >= 64; remaining -= 64) {
```

Comment:
This does not looks good. `size_t` is always better for loop counter and `remaining` looks unused for anything else.

Reply:
Why is `size_t` better for loop counter?

Reply:
It's natural register size. The `unsigned` in the other hand is 32-bit on 64-bit architecture and compiler has to guarantee wrapping behavior which is not always compatible with 64-bit architecture addressing modes.

In this case it does not make big difference, but I like to use it as a rule of thumb.

I cheated a bit by making `size_t len`, but the difference is mostly the ability to use `lea` instruction in this case.
https://godbolt.org/z/xM15GjPGd

Reply:
OK, interesting. I've change it to `size_t`.